core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/RowDenormalizerAggregator.java

```diff
@@ -128,6 +131,15 @@ private void validateInputFields(Schema inputSchema) {
     } else if ((inputSchema.getField(conf.getFieldValue())) == null) {
       throw new IllegalArgumentException(
         String.format("Fieldvalue '%s' does not exist in input schema %s", conf.getFieldValue(), inputSchema));
+    } else if (!inputSchema.getField(conf.getKeyField()).getSchema().getType().equals(Schema.Type.STRING)) {
+      throw new IllegalArgumentException(
+        String.format("Keyfield '%s' must be of type String", conf.getKeyField()));
+    } else if (!inputSchema.getField(conf.getFieldName()).getSchema().getType().equals(Schema.Type.STRING)) {
+      throw new IllegalArgumentException(
+        String.format("Fieldname '%s' must be of type String", conf.getFieldName()));
+    } else if (!inputSchema.getField(conf.getFieldValue()).getSchema().getType().equals(Schema.Type.STRING)) {
```

Comment:
seems like the decision is that the field value will be changed to a string regardless of what its original type is, which means you shouldn't be checking that the original type is a string.